### ISSUE TITLE
[CRASH] Crash when trying to go to the desired object standing in the factory from the mod Factorissimo 2 - notnotmelon fork 🍴

### ISSUE THREAD

**Issue by Eldrinn-Elantey** [tburrows13/FactorySearch#4]
```
8084.447 Error MainLoop.cpp:1284: Exception at tick 70603238: Мод Factory Search (Поиск по фабрике) (1.9.2) вызвал неустранимую ошибку.
Пожалуйста, сообщите об этой ошибке автору мода.

Error while running event FactorySearch::on_gui_click (ID 1)
__FactorySearch__/scripts/open_location.lua:129: attempt to index field 'building' (a nil value)
stack traceback:
	__FactorySearch__/scripts/open_location.lua:129: in function 'open_location'
	__FactorySearch__/scripts/gui.lua:693: in function <__FactorySearch__/scripts/gui.lua:666>
8084.447 Error ServerMultiplayerManager.cpp:92: MultiplayerManager failed: "Мод Factory Search (Поиск по фабрике) (1.9.2) вызвал неустранимую ошибку.
Пожалуйста, сообщите об этой ошибке автору мода.

Error while running event FactorySearch::on_gui_click (ID 1)
__FactorySearch__/scripts/open_location.lua:129: attempt to index field 'building' (a nil value)
stack traceback:
	__FactorySearch__/scripts/open_location.lua:129: in function 'open_location'
	__FactorySearch__/scripts/gui.lua:693: in function <__FactorySearch__/scripts/gui.lua:666>"
8084.447 Info ServerMultiplayerManager.cpp:797: updateTick(70603238) changing state from(InGame) to(Failed)
8084.448 Info SteamContext.cpp:486: Leaving lobby...

```

**Comment by tburrows13** [comment]
Thanks for the report. I think this crash can happen when the result is in a factory that doesn’t have a physical corresponding outside building anymore, e.g. it is in item form.

Should be straightforward for me to fix :)

**Comment by tburrows13** [comment]
Fixed in v1.9.3.
